test/python/transpiler/test_basic_swap.py

```diff
@@ -334,6 +343,359 @@ def test_far_swap_with_gate_the_middle(self):
 
         self.assertEqual(circuit_to_dag(expected), after)
 
+    @unittest.skipUnless(optionals.HAS_AER, "Qiskit Aer is required to run this test")
+    def test_controlflow_pre_if_else_route(self):
+        """test swap with if else controlflow construct"""
+        num_qubits = 5
+        qreg = QuantumRegister(num_qubits, "q")
+        creg = ClassicalRegister(num_qubits)
+        coupling = CouplingMap([(i, i + 1) for i in range(num_qubits - 1)])
+        qc = QuantumCircuit(qreg, creg)
+        qc.h(0)
+        qc.cx(0, 2)
+        qc.measure(2, 2)
+        true_body = QuantumCircuit(qreg, creg)
+        true_body.x(3)
+        false_body = QuantumCircuit(qreg, creg)
+        false_body.x(4)
+        qc.if_else((creg[2], 0), true_body, false_body, qreg, creg)
+        qc.measure(qreg, creg)
+
+        dag = circuit_to_dag(qc)
+        cdag = BasicSwap(coupling).run(dag)
+        cqc = dag_to_circuit(cdag)
+
+        sim = Aer.get_backend("aer_simulator")
+        in_results = sim.run(qc, shots=4096).result().get_counts()
+        out_results = sim.run(cqc, shots=4096).result().get_counts()
+        self.assertEqual(set(in_results), set(out_results))
```

Comment:
Do we have a utility function to test if a physical circuit is compliant with a given `coupling_map`? If not, this would be useful in verifying the outputs circuits here.

Reply:
For a target (which is superset of the coupling map) we have https://github.com/Qiskit/qiskit-terra/blob/main/test/python/providers/test_backend_v2.py#L43-L53 which I was using to validate the full `transpile()` pipe line was valid